sdk/storage/Azure.Storage.Common/src/Shared/StorageSharedKeyCredentialExtensions.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure.Storage
+{
+    internal class StorageSharedKeyCredentialExtensions : StorageSharedKeyCredential
```

Comment:
I'm kind of on the fence about calling this `Extensions` since there aren't any extension methods.  How do you feel about suffixing these with `Internals` instead to make the purpose clearer?  It'd also be good to add a comment that we're adding this type to access `protected static` members off the base class that we need available but don't want to directly expose to customers.

Reply:
I was thinking of classes suffixed with Extensions as containing helper methods that may also be extension methods. We have other Extension classes that contain non-extension methods, but it is true this is different as this doesn't have any extension methods.

Reply:
I think it's fine to have a class full of extension methods that we sometimes add other helpers to.  But this class will never have extension methods since it's not static.  I'd vote we change all these helpers to read `protected static` members `FooInternals` so the pattern is clear.